ui/components/src/test/tree/controlled/TreeNodeLoader.test.ts

```diff
@@ -618,4 +618,47 @@ describe("handleLoadedNodeHierarchy", () => {
     expect(modelSource.getModel().getChildren(parentNode.id)).to.be.undefined;
   });
 
+  it("updates existing expanded nodes in the same position", () => {
+    const root1 = createRandomTreeNodeItem("root1", undefined, "root-node-1");
+    const root2 = createRandomTreeNodeItem("root2", undefined, "root-node-2");
+    const root3 = createRandomTreeNodeItem("root3", undefined, "root-node-3");
+    const child1 = createRandomTreeNodeItem("child1", root2.id, "child-node-1");
+    const child2 = createRandomTreeNodeItem("child2", root3.id, "child-node-2");
+    const newNode = createRandomTreeNodeItem("new-root1", undefined, "new-root-node");
+    modelSource.modifyModel((model) => {
+      model.setChildren(
+        undefined,
+        [convertToTreeModelNodeInput(root1), { ...convertToTreeModelNodeInput(root2), isExpanded: true }, { ...convertToTreeModelNodeInput(root3), isExpanded: true, description: "description" }],
+        0,
+      );
+      model.setChildren(root2.id, [convertToTreeModelNodeInput(child1)], 0);
+      model.setChildren(root3.id, [convertToTreeModelNodeInput(child2)], 0);
+    });
+
+    // numChildren set to undefined indicates that this is not he first request for children response

```

Comment:
Please move this comment to `TreeNodeLoader.ts:352` and remove `istanbul ignore` since this test now covers that case.